blog-content/articles/protect-your-future-self-by-encapsulating-test-input-preparation.md

```diff
@@ -0,0 +1,127 @@
+---
+date: 2020-01-07
+summary: Examples of how to reduce verbosity and brittleness of your tests by encapsulating input preparation 
+---
+
+# Tests Brittleness and Verbosity slows you down
+
+In [Testing Makes You Faster Day one](entry/testing-makes-you-faster-day-one) article I claimed that writing tests makes you faster.
+However there are test patterns that most likely will slow you down and you need to be ready to handle them. 
+
+Writing tests for a simple logic with a simple input is a breeze. Take for example a test for a simple calculator class. 
+Two numeric inputs and we are done: 
+
+:include-file: com/example/portfolio/SimpleCalculatorTest.java {title: "SimpleCalculatorTest.java"}
+
+However real apps require real domain, and real domain is rarely represented by a couple of numbers.
+Let's take Financial domain as an example. It is full of complex types such as `Transaction` and has a lot of business logic.
+
+One example of such business logic is `ProfitCalculator`. Its job is to calculate profit based on the executed `Transactions`.
+
+After consulting with the business people we are ready to give our implementation a shot.
+
+`Transaction` is a simple data class with a few fields (number of fields is reduced for simplicity).
```

Comment:
I would take the word simple out, because we called 'Transaction' complex earlier in the article :). Alternatively, I would change the sentence to say that we simplified it by reducing some fields.